### DISCUSSION TITLE
Is there an ``onBeforeDestroy()`` callback?

### DISCUSSION THREAD

**Discussion by KiddoV** [jcubic/jquery.terminal#917] [Q&A]
Is there any like ``onBeforeDestroy()`` callback or something similar to call before terminal got destroy ``term.destroy()``?
Thanks,

**Comment by jcubic** [comment]
No, I'm afraid. But it's good idea to add also `onAfterDestroy()`.

To add event you only need to add:

```javascript
fire_event('onBeforeDestroy', [])
```

and probably good idea to check if the value is false and don't execute `destroy()`.

```javascript
if (fire_event('onBeforeDestroy', []) === false) {
   return this;
}
```

**Comment by KiddoV** [comment]
I am wrapping `jquery.terminal` in a Svelte component, can you point me to the right direction of where should I call `fire_event` method?
```ts
import { onMount, onDestroy } from 'svelte';

export let termIns: JQueryTerminal = Object.create(null);
let jqTermContainerEl: HTMLElement;
let termDefaultOpts: JQueryTerminal.TerminalOptions;

onMount(() => {
    termIns = jQuery(jqTermContainerEl).terminal(termInterp, {...termDefaultOpts, ...termOpts});
});

onDestroy(() => {
    //Should we call it here before termIns.destroy()?
    termIns.fire_event('onBeforeDestroy', []) // <= ERR: fire_event is not a function
    if (termIns) termIns.purge().destroy();
});
```
Thanks,

**Reply by jcubic** [reply]
No, the code `fire_event` is supposed to go inside jQuery Terminal source code. This not a public API. It needs to be added inside destroy method.

**Reply by KiddoV** [reply]
I will be waiting for the new update then, if you ever plan to add this hook.
